### PR TITLE
[docs] Add Google authentication with AuthSession API guide

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Closes ENG-7132

# How

This PR adds a step-by-step guide on using Google as an authentication provider with AuthSession API. The objective of the guide is to provide is to cover the steps to generate credentials for different platforms, provide a minimal working example, list out different use cases (development builds and production apps), and guide through how the AuthSession API works with a real-world example.

The guide is currently placed under "Integrations".

# Test Plan

Changes have been tested locally by visiting http://localhost:3002/guides/google-authentication/.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/guides/google-authentication.mdx [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+  text: {
+    fontSize: 20,
+    fontWeight: "bold",
+  },
+});
+```
+
+## Testing your app
+
+**For a development build**, see [Use development builds](/development/use-development-builds/#start-the-development-server) for further instructions on how to start the development server or run [`npx expo run:android` or `npx expo run:ios`](/workflow/expo-cli/#compiling) command to test it locally.
```
Comment (amandeepmittal) [inline docs/pages/guides/google-authentication.mdx:277]:
```suggestion
**For a development build**, see [Use development builds](/development/use-development-builds/#start-the-development-server) for further instructions on how to start the development server or run [`npx expo run:android` or `npx expo run:ios`](/workflow/expo-cli/#compiling) command to test the native project it locally.
```

File: docs/pages/guides/google-authentication.mdx [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+  text: {
+    fontSize: 20,
+    fontWeight: "bold",
+  },
+});
+```
+
+## Testing your app
+
+**For a development build**, see [Use development builds](/development/use-development-builds/#start-the-development-server) for further instructions on how to start the development server or run [`npx expo run:android` or `npx expo run:ios`](/workflow/expo-cli/#compiling) command to test the native project it locally.
```
Comment (amandeepmittal) [inline docs/pages/guides/google-authentication.mdx:277]:
```suggestion
**For a development build**, see [Use development builds](/development/use-development-builds/#start-the-development-server) for further instructions on how to start the development server or run [`npx expo run:android` or `npx expo run:ios`](/workflow/expo-cli/#compiling) command to test the native project locally.
```

File: docs/pages/guides/google-authentication.mdx [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+- In the Google Cloud console, go to **Credentials** > **Create OAuth client** page.
+- For **Name**, enter the name such as **Web client ID** to identify different client IDs in case you have more than one for multiple platforms.
+- For **Authorized JavaScript origins**, add the URI: `https://localhost:19006` and `https://yourwebsite.com`.
+- For **Authorized redirect URIs**, add the URI: `https://localhost:19006` and `https://yourwebsite.com`.
+
+</Tab>
+
+</Tabs>
+
+## Using AuthSession API
```
Comment (amandeepmittal) [inline docs/pages/guides/google-authentication.mdx:112]:
```suggestion
## Use AuthSession API
```

File: docs/pages/guides/google-authentication.mdx [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+
+<Terminal cmd={['$ npx uri-scheme add your-ios.bundleIdentifier --ios']} />
+
+</Tab>
+
+</Tabs>
+
+</Collapsible>
+
+## More
```
Comment (Simek) [inline docs/pages/guides/google-authentication.mdx:315]:
```suggestion
## Learn more
```

Small nit.

**Review by Simek — APPROVED** [review]
Thank you for writing this guide! LGTM 🚀

**Comment by amandeepmittal** [comment]
Changes made to the PR:
- remove Expo Go references
- Move Web to its own section (as suggested by Brent)

File: docs/pages/guides/google-authentication.mdx [hunk 5]
```diff
@@ -0,0 +1,319 @@
+---
+title: Authentication with Google and AuthSession API
+sidebar_title: Using Google Authentication
```
Comment (amandeepmittal) [inline docs/pages/guides/google-authentication.mdx:3]:
```suggestion
sidebar_title: Using Google authentication
```

**Review by brentvatne — APPROVED** [review]

**Pull request merged**